packages/mui-material/src/Avatar/Avatar.test.js

```diff
@@ -212,4 +213,28 @@ describe('<Avatar />', () => {
       expect(avatar).to.have.class(classes.colorDefault);
     });
   });
+
+  it('should not error when ownerState is used in styleOverrides', () => {
+    const theme = createTheme({
+      components: {
+        MuiAvatar: {
+          styleOverrides: {
+            root: ({ ownerState }) => ({
+              ...(ownerState.variant === 'rounded' && {
+                color: 'red',
+              }),
+            }),
+          },
+        },
+      },
+    });
+
+    expect(() =>
+      render(
+        <ThemeProvider theme={theme}>
+          <Avatar variant="rounded" />
+        </ThemeProvider>,
+      ),
+    ).not.toErrorDev();
```

Comment:
`toErrorDev` is used when you want to assert `console.error` messages. We should use `throw()` here:
```suggestion
    ).not.to.throw();
```

Reply:
Oh okay this is new to me. I've made changes as suggested